baybe/campaign.py

```diff
@@ -222,6 +265,72 @@ def add_measurements(
             numerical_measurements_must_be_within_tolerance,
         )
 
+    def _mark_as_measured(
+        self,
+        measurements: pd.DataFrame,
+        numerical_measurements_must_be_within_tolerance: bool,
+    ) -> None:
+        """Mark the given elements of the discrete subspace as measured.
+
+        Args:
+            measurements: A dataframe containing parameter configurations to be
+                marked as measured.
+            numerical_measurements_must_be_within_tolerance: See
+                :func:`baybe.utils.dataframe.fuzzy_row_match`.
+        """
+        idxs_matched = fuzzy_row_match(
+            self.searchspace.discrete.exp_rep,
+            measurements,
+            self.parameters,
+            numerical_measurements_must_be_within_tolerance,
+        )
+        self._searchspace_metadata.loc[idxs_matched, _MEASURED] = True
+
+    def toggle_discrete_candidates(
+        self,
+        filter: pd.DataFrame,
+        exclude: bool,
+        anti: bool = False,
+        dry_run: bool = False,
+    ) -> pd.DataFrame:
+        """Un-/exclude certain discrete points from the candidate set.
+
+        Args:
+            filter: A dataframe specifying the filtering mechanism to determine the
```

Comment:
I also had a hard time finding a good name here and there might be better alternatives. Happy to hear your (@AVHopp, @Scienfitz) suggestions. The reason why "candidates" or "points" doesn't work is because you can pass incomplete/partial candidate specification here. See for example the newly introduced test, where the searchspace consists of two parameters but you define your "filtering" mechanism only based on values of the first.

So taking that into consideration, what do you think would fit better?

Reply:
Hmm.... maybe then something like `values_to_filter`? Although this is also not really correct and I'm not sure if this is better. If you prefer that name feel free to change, if not feel free to resolve as I agree that finding another name is probably hard.

Reply:
Have found the solution I think. Will explain the changes in another thread in a minute. But as summary here: it's now called `constraints`, which is a consistent theme across our entire codebase that will be expanded even further soon